src/InlineAutocomplete/InlineAutocomplete.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {cloneElement, useRef} from 'react'
+import Box from '../Box'
+import {useCombinedRefs} from '../hooks/useCombinedRefs'
+import {useSyntheticChange} from '../hooks/useSyntheticChange'
+import {BetterSystemStyleObject} from '../sx'
+
+import {ShowSuggestionsEvent, Suggestions, TextInputCompatibleChild, TextInputElement, Trigger} from './types'
+import {
+  augmentHandler,
+  calculateSuggestionsQuery,
+  getAbsoluteCharacterCoordinates,
+  requireChildrenToBeInput
+} from './utils'
+import AutocompleteSuggestions from './_AutocompleteSuggestions'
+
+export interface InlineAutocompleteProps {
+  /** Register the triggers that can cause suggestions to appear. */
+  triggers: Array<Trigger>
+  /**
+   * Called when a valid suggestion query is updated. This should be handled by setting the
+   * `suggestions` prop accordingly.
+   */
+  onShowSuggestions: (event: ShowSuggestionsEvent) => void
+  /** Called when suggestions should be hidden. Set `suggestions` to `null` in this case. */
+  onHideSuggestions: () => void
+  /**
+   * The currently visible list of suggestions. If `loading`, a loading indicator will be
+   * shown. If `null` or empty, the list will be hidden. Suggestion sort will be preserved.
+   *
+   * Typically, this should not contain more than five or so suggestions.
+   */
+  suggestions: Suggestions | null
+  /**
+   * The `AutocompleteTextarea` has a container for positioning the suggestions overlay.
+   * This can break some layouts (ie, if the editor must expand with `flex: 1` to fill space)
+   * so you can override container styles here. Usually this should not be necessary.
+   * `position` may not be overriden.
+   */
+  sx?: Omit<BetterSystemStyleObject, 'position'>
+  // Typing this as such makes it look like a compatible child internally, but it isn't actually
+  // enforced externally so we have to resort to a runtime assertion.
+  /**
+   * An `input` or `textarea` compatible component to extend. A compatible component is any
+   * component that forwards a ref and props to an underlying `input` or `textarea` element,
+   * including but not limited to `Input`, `TextArea`, `input`, `textarea`, `styled.input`,
+   * and `styled.textarea`. If the child is not compatible, a runtime `TypeError` will be
+   * thrown.
+   */
+  children: TextInputCompatibleChild
+}
+
+const getSelectionStart = (element: TextInputElement) => {
+  try {
+    return element.selectionStart
+  } catch (e: unknown) {
+    // Safari throws an exception when trying to access selectionStart on date input element
+    if (e instanceof TypeError) return null
+    throw e
+  }
+}
+
+const noop = () => {
+  // don't do anything
+}
+
+/**
+ * Shows suggestions to complete the current word/phrase the user is actively typing.
+ */
+const InlineAutocomplete = ({
+  triggers,
+  suggestions,
+  onShowSuggestions,
+  onHideSuggestions,
+  sx,
+  children,
+  // Forward accessibility props so it works with FormControl
+  ...forwardProps
```

Comment:
The `FormControl` component clones its child and adds an `id` prop to it. Since the child in this case will be `InlineAutocomplete` instead of `Input`, we need to forward the ID on down.